### PR TITLE
Add prompts endpoints

### PR DESCRIPTION
**Pull request by unhappychoice**
### Changes

Add prompts endpoint.

##### `GET /api/v2/prompts`
- `Auth0Client#prompts`
- `Auth0Client#get_prompts`

##### `PATCH /api/v2/prompts`
- `Auth0Client#update_prompts`
- `Auth0Client#patch_prompts`

##### `GET /api/v2/prompts/{prompt}/custom-text/{language}`
- `Auth0Client#custom_text`
- `Auth0Client#get_custom_text`

##### `PUT /api/v2/prompts/{prompt}/custom-text/{language}`
- `Auth0Client#update_custom_text`
- `Auth0Client#put_custom_text`

### References

Closes #205 

- https://auth0.com/docs/api/management/v2#!/Prompts/get_prompts
- https://auth0.com/docs/api/management/v2#!/Prompts/patch_prompts
- https://auth0.com/docs/api/management/v2#!/Prompts/get_custom_text_by_language
- https://auth0.com/docs/api/management/v2#!/Prompts/put_custom_text_by_language

### Testing

* [x] This change adds unit test coverage
* [ ] This change adds integration test coverage
* [ ] This change has been tested on the latest version of Ruby

### Checklist

* [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
* [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
* [x] All existing and new tests complete without errors
* [ ] Rubocop passes on all added/modified files => Lots of errors in my environment. 🤔 
* [ ] All active GitHub checks have passed

### CODE REVIEW

**Comment by unhappychoice** [comment]
I just noticed there are integration tests.
So I will manage that and make sure the methods are actually working in my application using auth0.

**Comment by unhappychoice** [comment]
> I just noticed there are integration tests. So, I will manage that
I tried but I can't because I don't have `ENV['MASTER_JWT']` or other authentication information. ( Is this right? )

> make sure the methods are actually working in my application using auth0 from now on.

I confirmed that the methods are working in my test auth0 application.

**Comment by qortex** [comment]
What is the status of this PR, are you positive on a merge soon?

**Comment by unhappychoice** [comment]
> What is the status of this PR, are you positive on a merge soon?
Yes, I am.
@damieng @joshcanhelp

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      expect { @instance.prompts }.not_to raise_error
+    end
+  end
+
+  context '.update_prompts' do
+    it 'is expected to respond to a prompts method' do
+      expect(@instance).to respond_to(:update_prompts)
+    end
+
+    it 'is expected to respond to a get_prompts method' do
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:30]:
```suggestion
    it 'is expected to respond to a patch_prompts method' do
```

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    it 'is expected to get /api/v2/prompts' do
+      expect(@instance).to receive(:get).with('/api/v2/prompts')
+
+      expect { @instance.prompts }.not_to raise_error
+    end
+  end
+
+  context '.update_prompts' do
+    it 'is expected to respond to a prompts method' do
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:26]:
```suggestion
    it 'is expected to respond to an update_prompts method' do
```

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    it 'is expected to call patch /api/v2/prompts' do
+      expect(@instance).to receive(:patch).with('/api/v2/prompts', { universal_login_experience: 'new' })
+
+      expect { @instance.update_prompts(universal_login_experience: 'new') }.not_to raise_error
+    end
+  end
+
+  context '.custom_text' do
+    it 'is expected to respond to a prompts method' do
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:42]:
```suggestion
    it 'is expected to respond to a custom_text method' do
```

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      expect(@instance).to respond_to(:custom_text)
+    end
+
+    it 'is expected to respond to a get_prompts method' do
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:46]:
```suggestion
    it 'is expected to respond to a get_custom_text method' do
```

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    end
+
+    it 'is expected to get /api/v2/prompts/PROMPT_NAME/language/LANGUAGE' do
+      expect(@instance).to receive(:get).with('/api/v2/prompts/PROMPT_NAME/custom-text/LANGUAGE')
+
+      expect { @instance.custom_text('PROMPT_NAME', 'LANGUAGE') }.not_to raise_error
+    end
+
+    it 'is expected to raise an exception when the prompt is empty' do
+      expect { @instance.custom_text(nil, nil) }.to raise_exception(Auth0::InvalidParameter)
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:57]:
```suggestion
      expect { @instance.custom_text(nil, 'language') }.to raise_exception(Auth0::InvalidParameter)
```

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    end
+
+    it 'is expected to raise an exception when the prompt is empty' do
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:60]:
```suggestion
    it 'is expected to raise an exception when the language is empty' do
```

File: lib/auth0/api/v2/prompts.rb [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+          get(prompts_path)
+        end
+        alias get_prompts prompts
+
+        # Update prompts settings.
+        # @see https://auth0.com/docs/api/management/v2#!/Prompts/patch_prompts
+        # @param options [hash]
+        #   * :universal_login_experience [string] Should be any of: new, classic.
+        #
+        # @return [json] Returns the prompts setting.
```
Comment (Widcket) [inline lib/auth0/api/v2/prompts.rb:20]:
```suggestion
        # @return [json] Returns the prompts settings.
```

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+      expect { @instance.custom_text('prompt_name', nil) }.to raise_exception(Auth0::InvalidParameter)
+    end
+  end
+
+  context '.update_custom_text' do
+    it 'is expected to respond to a prompts method' do
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:66]:
```suggestion
    it 'is expected to respond to an update_custom_text method' do
```

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+      expect(@instance).to respond_to(:update_custom_text)
+    end
+
+    it 'is expected to respond to a get_prompts method' do
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:70]:
```suggestion
    it 'is expected to respond to a put_custom_text method' do
```

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+      expect(@instance).to respond_to(:put_custom_text)
+    end
+
+    it 'is expected to get /api/v2/prompts/PROMPT_NAME/language/LANGUAGE' do
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:74]:
```suggestion
    it 'is expected to put /api/v2/prompts/PROMPT_NAME/language/LANGUAGE' do
```

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+      expect(@instance).to receive(:put).with('/api/v2/prompts/PROMPT_NAME/custom-text/LANGUAGE', 'BODY')
+
+      expect { @instance.update_custom_text('PROMPT_NAME', 'LANGUAGE', 'BODY') }.not_to raise_error
+    end
+
+    it 'is expected to raise an exception when the prompt is empty' do
+      expect { @instance.update_custom_text(nil, nil, 'BODY') }.to raise_exception(Auth0::InvalidParameter)
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:81]:
```suggestion
      expect { @instance.update_custom_text(nil,'language', 'BODY') }.to raise_exception(Auth0::InvalidParameter)
```

File: spec/lib/auth0/api/v2/prompts_spec.rb [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+    end
+
+    it 'is expected to raise an exception when the prompt is empty' do
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/prompts_spec.rb:84]:
```suggestion
    it 'is expected to raise an exception when the language is empty' do
```

**Comment by Widcket** [comment]
This is out now on [4.13.0](https://rubygems.org/gems/auth0/versions/4.13.0).

**Pull request merged**